src/Paragram.jl

```diff
@@ -0,0 +1,190 @@
+
+            @info("Downloading",
+                  source=url,
+                  dest = filepath,
+                  progress = completion_progress |> format_progress,
+                  time_taken = taken_time |> format_seconds,
+                  time_remaining = remaining_time |> format_seconds,
+                  average_speed = average_speed |> format_bytes_per_second,
+                  downloaded = downloaded_bytes |> format_bytes,
+                  remaining = remaining_bytes |> format_bytes,
+                  total = total_bytes |> format_bytes,
+                  )
+        end
+
+
+        Base.open(filepath, "w") do fh
+            while(!eof(stream))
+                downloaded_bytes += write(fh, readavailable(stream))
+                if !isinf(period)
+                  if now() - prev_time > Millisecond(1000*period)
+                    report_callback()
+                  end
+                end
+            end
+        end
+        report_callback()
+    end
+    filepath
+end
+#const Paragram_max_size = 4_000_000
+
+function init(::Type{Paragram})
+    vectors = [("paragram_300_ws353",
+                " Paragram-WS353 300 dimensional Paragram embeddings tuned on WordSim353 dataset. 1.7 GB download.",
+                "8ed9a19f8bc400cdbca5dae7f024c0310a3e5711a46ba48036a7542614440721",
+              "https://drive.google.com/uc?id=0B9w48e1rj-MOLVdZRzFfTlNsem8&export=download"),
+                # "https://drive.google.com/file/d/0B9w48e1rj-MOLVdZRzFfTlNsem8/view"),
+               ("paragram_300_sl999",
+                "Paragram-SL999 300 dimensional Paragram embeddings tuned on SimLex999 dataset. 1.7 GB download.",
+                "9a16adc7d620642f863278451db4c03a2646016440ccea7e30a37ba17868781d",
+               "https://drive.google.com/uc?id=0B9w48e1rj-MOck1fRGxaZW1LU2M&export=download"),
+]
+#https://drive.google.com/file/d/0B9w48e1rj-MOLVdZRzFfTlNsem8/view?usp=sharing
+    for (depname, description, sha, link) in vectors
+        register(DataDep(depname,
+                         """
+                         Pretrained Paragram word embeddings.
+                         Website: https://www.cs.cmu.edu/~jwieting/
+                         Author: John Wieting
+                         Year: 2015
+                         Licence: Open Data Commons Public Domain Dedication and License (PDDL)
+
+                         Paper:
+                             John Wieting, Mohit Bansal, Kevin Gimpel, Karen Livescu, Dan Roth. From Paraphrase Database to Compositional Paraphrase Model and Back.
+
+                         $description
+                         """,
+                         link,
+                         sha,
+                         fetch_method = download_gdrive))
+       
+            append!(language_files(Paragram{:en}), ["$(depname)/$(depname).txt"])
+       
+            
+      
+    end
+end
+
+function _load_embeddings(::Type{<:Paragram}, embedding_file, max_vocab_size, keep_words)
```

Comment:
Is this literally identical to some existing code we have?
E.g. in FastText and Glove?
Shall we factor it out into a new `common.jl`
and call it `_load_embeddings_csv(embedding_file, max_vocab_size, keep_words)`
and then just have `_load_embeddings(::Type{<:Paragram}, args...) = _load_embeddings_csv(args)`  etc?

Reply:
yeah the `_load_embeddings_csv(embedding_file, max_vocab_size, keep_words)` in 'Paragram.jl' is exactly same as in Glove and similar to FastText 
Yeah I think it is good idea to call `common.jl`. If you alIow me,I would love to do it